distributed/utils_comm.py

```diff
@@ -77,36 +86,42 @@ async def gather_from_workers(
                         address,
                         who=who,
                         serializers=serializers,
-                        max_connections=False,
                     ),
                     operation="get_data_from_worker",
                 ),
                 name=f"get-data-from-{address}",
             )
             for address, keys in d.items()
         }
-        response: dict[str, object] = {}
-        for worker, c in coroutines.items():
+        for address, task in tasks.items():
             try:
-                r = await c
+                r = await task
```

Comment:
I know this is the same as before but the task scheduling pattern here is problematic in the event of cancellation.

If `gather_from_workers` is cancelled after the tasks were created, only the task we're currently awaiting is cancelled, all others will continue running and we'll get a "never awaited foo" warning.

I _guess_ this coroutinefunction is never actually cancelled which is why we never ran into this...

The _correct_ approach to this would be to use `asyncio.gather` (or even better but not backwards copatible, the [TaskGroup of asyncio in python 3.11+](https://docs.python.org/3/library/asyncio-task.html#asyncio.TaskGroup))

I think the changes should be straight forward to something like

```python
results = asyncio.gather(tasks, return_exceptions=True)
for addr, res in results.items():
    if isinstance(res, OSError):
        ...
```

Reply:
I'll address it in a follow-up